recipes/libxmlpp/2.x.x/conanfile.py

```diff
@@ -26,130 +31,114 @@ class LibXMLPlusPlus(ConanFile):
         "shared": False,
         "fPIC": True,
     }
-    generators = "pkg_config"
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
 
     @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def _lib_version(self):
+        return "2.6" if Version(self.version) <= "2.42.1" else "5.0"
 
     def export_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("libxml2/2.9.14")
-        if tools.Version(self.version) <= "2.42.1":
-            self.requires("glibmm/2.66.4")
+        self.requires("libxml2/2.11.4")
+        if Version(self.version) <= "2.42.1":
+            self.requires("glibmm/2.66.4", transitive_headers=True, transitive_libs=True)
         else:
-            self.requires("glibmm/2.72.1")
+            self.requires("glibmm/2.75.0")
 
     def validate(self):
-        if hasattr(self, "settings_build") and tools.cross_building(self):
+        if hasattr(self, "settings_build") and cross_building(self):
             raise ConanInvalidConfiguration("Cross-building not implemented")
-
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration(f"{self.name} recipe is not yet available for Macos")
```

Comment:
```suggestion
        if Version(self.version) < "2.91.1":
            from conan import conan_version
            import sys
            if conan_version.major == 2:
                # FIXME: linter complains, but function is there
                # https://docs.conan.io/2.0/reference/tools/build.html?highlight=check_min_cppstd#conan-tools-build-check-max-cppstd
                check_max_cppstd = getattr(sys.modules['conan.tools.build'], 'check_max_cppstd')
                # INFO: error: no template named 'auto_ptr' in namespace 'std'. Removed in C++17.
                check_max_cppstd(self, 14)
```
That error is because auto_ptr was excluded from C++ standard library, and Clang is using C++17. It's already fixed, so we should not block newer versions.

More info: https://github.com/libxmlplusplus/libxmlplusplus/blob/5f5371776186fb3390d028848bed10953d17a865/NEWS#L201